lib/charms/operator_libs_linux/v2/snap.py

```diff
@@ -580,10 +580,15 @@ def ensure(
             # We are installing or refreshing a snap.
             if self._state not in (SnapState.Present, SnapState.Latest):
                 # The snap is not installed, so we install it.
+                logger.info(
+                    f"Installing snap {self._name} revision {revision}, tracking {channel}"
```

Comment:
Best practice is to not use f-strings in log messages. For example, some logging handlers will use the raw string for de-duping/grouping. Same for the below as well.

```suggestion
                    "Installing snap %s revision %s, tracking %s", self._name, revision, channel
```

Reply:
TIL, thanks!